Makefile

```diff
@@ -213,17 +216,17 @@ endif
 
 .PHONY: verify-lint
 verify-lint: golangci
-	golangci-lint run --timeout 10m --modules-download-mode=readonly || (echo 'Run "make update"' && exit 1)
+	$(GOPATH)/bin/golangci-lint run --timeout 10m --modules-download-mode=readonly || (echo 'Run "make update"' && exit 1)
 
 .PHONY: update-lint
 update-lint: golangci
-	golangci-lint run --fix --modules-download-mode=readonly
+	$(GOPATH)/bin/golangci-lint run --fix --modules-download-mode=readonly
 
-HAS_GOLANGCI:=$(shell which golangci-lint)
+HAS_GOLANGCI_VERSION:=$(shell $(GOPATH)/bin/golangci-lint version --format=short)
 .PHONY: golangci
 golangci:
-ifndef HAS_GOLANGCI
-	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.48.0
+ifneq ($(HAS_GOLANGCI_VERSION), $(GOLANGCI_VERSION))
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(GOPATH)/bin v$(GOLANGCI_VERSION)
```

Comment:
Replacing `$(go env GOPATH)/bin` with `$(GOPATH)/bin` can lead to unexpected results. Difference between them is that `$GOPATH` is en environment variable that can be empty and leading to aforementioned path to be resolved to `/bin`. This is not wanted as will lead to user confusion when they get permission error for accessing `/bin`.

 `$(go env GOPATH)` will not only return `$GOPATH` environment value, but also will provide a fallback based on localtion of go binary.

Reply:
@serathius Actually, in the context of this Makefile:
- `$GOPATH` is **not** an environment variable, but a Make variable, whose value is the output of `go env GOPATH` (initialized [line 23](https://github.com/kubernetes-sigs/metrics-server/blob/0556b85536db201fe82558a99cff679596c8b161/Makefile#L23)). It is therefore **not empty**.
- On the contrary, `$(go env GOPATH)` resolves to `""` ($ is for Make variables); `$$(go env GOPATH)` would work ($$ is for shell variables).

Here is a simple Makefile to show it better than words:
```make
GOPATH:=$(shell go env GOPATH)

.PHONY: var
var:
	echo $(GOPATH)
	# output: my local GOPATH path

.PHONY: one
one:
	echo $(go env GOPATH)
	# output: nothing

.PHONY: two
two:
	echo $$(go env GOPATH)
	# output: my local GOPATH path
```

Reply:
Missed that. Thanks for detailed explanation.